sycl/source/detail/event_impl.cpp

```diff
@@ -210,15 +210,17 @@ void event_impl::wait(
 
 void event_impl::wait_and_throw(
     std::shared_ptr<cl::sycl::detail::event_impl> Self) {
+  Scheduler &Sched = Scheduler::getInstance();
+  Scheduler::ReadLockT Lock(Sched.MGraphLock);
+
   Command *Cmd = static_cast<Command *>(Self->getCommand());
   QueueImplPtr submittedQueue = nullptr;
   if (Cmd)
     submittedQueue = Cmd->getSubmittedQueue();
 
   wait(Self);
```

Comment:
Call to `wait` should not take place with graph lock taken.

Reply:
Fixed it